### PR TITLE
Update SQL Bridge Sample

### PR DESCRIPTION
**Pull request by petersgiles**
relates to: https://github.com/Particular/docs.particular.net/issues/1010

### CODE REVIEW

**Comment by SimonCropp** [comment]
it this a WIP?

**Comment by petersgiles** [comment]
@SimonCropp says so in the title, is there something else I should do?

**Comment by SimonCropp** [comment]
@petersgiles sorry. i typed that in the wrong issue

**Comment by petersgiles** [comment]
@indualagarsamy @SimonCropp said you might be working on this. I've taken the samples up to use the async startup pattern but need help understanding the changes to the api in the MsmqReceiver class.

**Comment by petersgiles** [comment]
```SQL
USE [PersistenceForMsmqTransport]
GO

INSERT INTO [dbo].[Subscription]
           ([SubscriberEndpoint]
           ,[MessageType]
           ,[LogicalEndpoint]
           ,[Version]
           ,[TypeName])
     VALUES
           (
'MsmqToSqlBridge@localhost',	'Shared.SomethingHappened,0.0.0.0',	'MsmqToSqlBridge'	,'0.0.0.0',	'Shared.SomethingHappened')
GO
```

**Comment by petersgiles** [comment]
```SQL
USE [master]
GO

/****** Object:  Database [PersistenceForMsmqTransport]    Script Date: 4/02/2016 2:45:31 PM ******/
CREATE DATABASE [PersistenceForMsmqTransport]
 CONTAINMENT = NONE
 ON  PRIMARY 
( NAME = N'PersistenceForMsmqTransport', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForMsmqTransport.mdf' , SIZE = 5120KB , MAXSIZE = UNLIMITED, FILEGROWTH = 1024KB )
 LOG ON 
( NAME = N'PersistenceForMsmqTransport_log', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForMsmqTransport_log.ldf' , SIZE = 1024KB , MAXSIZE = 2048GB , FILEGROWTH = 10%)
GO

ALTER DATABASE [PersistenceForMsmqTransport] SET COMPATIBILITY_LEVEL = 120
GO

```

**Comment by petersgiles** [comment]
```SQL
USE [master]
GO

/****** Object:  Database [PersistenceForSqlTransport]    Script Date: 4/02/2016 2:46:48 PM ******/
CREATE DATABASE [PersistenceForSqlTransport]
 CONTAINMENT = NONE
 ON  PRIMARY 
( NAME = N'PersistenceForSqlTransport', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForSqlTransport.mdf' , SIZE = 5120KB , MAXSIZE = UNLIMITED, FILEGROWTH = 1024KB )
 LOG ON 
( NAME = N'PersistenceForSqlTransport_log', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForSqlTransport_log.ldf' , SIZE = 1024KB , MAXSIZE = 2048GB , FILEGROWTH = 10%)
GO

ALTER DATABASE [PersistenceForSqlTransport] SET COMPATIBILITY_LEVEL = 120
GO
```

**Comment by petersgiles** [comment]
With Tables

```SQL
USE [master]
GO

/****** Object:  Database [PersistenceForMsmqTransport]    Script Date: 4/02/2016 2:51:40 PM ******/
DROP DATABASE [PersistenceForMsmqTransport]
GO

/****** Object:  Database [PersistenceForSqlTransport]    Script Date: 4/02/2016 2:52:00 PM ******/
DROP DATABASE [PersistenceForSqlTransport]
GO

/****** Object:  Database [PersistenceForSqlTransport]    Script Date: 4/02/2016 2:46:48 PM ******/
CREATE DATABASE [PersistenceForSqlTransport]
 CONTAINMENT = NONE
 ON  PRIMARY 
( NAME = N'PersistenceForSqlTransport', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForSqlTransport.mdf' , SIZE = 5120KB , MAXSIZE = UNLIMITED, FILEGROWTH = 1024KB )
 LOG ON 
( NAME = N'PersistenceForSqlTransport_log', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForSqlTransport_log.ldf' , SIZE = 1024KB , MAXSIZE = 2048GB , FILEGROWTH = 10%)
GO

ALTER DATABASE [PersistenceForSqlTransport] SET COMPATIBILITY_LEVEL = 120
GO



/****** Object:  Database [PersistenceForMsmqTransport]    Script Date: 4/02/2016 2:45:31 PM ******/
CREATE DATABASE [PersistenceForMsmqTransport]
 CONTAINMENT = NONE
 ON  PRIMARY 
( NAME = N'PersistenceForMsmqTransport', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForMsmqTransport.mdf' , SIZE = 5120KB , MAXSIZE = UNLIMITED, FILEGROWTH = 1024KB )
 LOG ON 
( NAME = N'PersistenceForMsmqTransport_log', FILENAME = N'C:\Program Files\Microsoft SQL Server\MSSQL12.SQLEXPRESS\MSSQL\DATA\PersistenceForMsmqTransport_log.ldf' , SIZE = 1024KB , MAXSIZE = 2048GB , FILEGROWTH = 10%)
GO

ALTER DATABASE [PersistenceForMsmqTransport] SET COMPATIBILITY_LEVEL = 120
GO

USE [PersistenceForMsmqTransport]
GO
/****** Object:  Table [dbo].[Subscription]    Script Date: 4/02/2016 2:54:55 PM ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
SET ANSI_PADDING ON
GO
CREATE TABLE [dbo].[Subscription](
	[SubscriberEndpoint] [varchar](450) NOT NULL,
	[MessageType] [varchar](450) NOT NULL,
	[LogicalEndpoint] [varchar](450) NULL,
	[Version] [varchar](450) NULL,
	[TypeName] [varchar](450) NULL,
PRIMARY KEY CLUSTERED 
(
	[SubscriberEndpoint] ASC,
	[MessageType] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]

GO
SET ANSI_PADDING OFF
GO
/****** Object:  Table [dbo].[TimeoutEntity]    Script Date: 4/02/2016 2:54:55 PM ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
SET ANSI_PADDING ON
GO
CREATE TABLE [dbo].[TimeoutEntity](
	[Id] [uniqueidentifier] NOT NULL,
	[Destination] [nvarchar](1024) NULL,
	[SagaId] [uniqueidentifier] NULL,
	[State] [varbinary](max) NULL,
	[Time] [datetime] NULL,
	[Headers] [nvarchar](max) NULL,
	[Endpoint] [nvarchar](440) NULL,
PRIMARY KEY NONCLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]

GO
SET ANSI_PADDING OFF
GO


INSERT INTO [dbo].[Subscription] ([SubscriberEndpoint] ,[MessageType] ,[LogicalEndpoint] ,[Version] ,[TypeName])
     VALUES
           ('MsmqToSqlBridge@localhost', 'Shared.SomethingHappened,0.0.0.0', 'MsmqToSqlBridge' ,'0.0.0.0', 'Shared.SomethingHappened')
GO
```

**Comment by indualagarsamy** [comment]
@petersgiles - I am going to rename the projects to `SqlRelay` instead of `SqlBridge`
Can you change your script to

INSERT INTO [dbo].[Subscription] ([SubscriberEndpoint] ,[MessageType] ,[LogicalEndpoint] ,[Version] ,[TypeName])
     VALUES
           ('MsmqToSqlRelay@localhost', 'Shared.SomethingHappened,0.0.0.0', 'MsmqToSqlRelay' ,'0.0.0.0', 'Shared.SomethingHappened')

**Comment by petersgiles** [comment]
@SimonCropp please review, we think it's OK

**Comment by indualagarsamy** [comment]
@SimonCropp - Once I'm done with the more native (MSMQ-> SQL) sample, I'll create that as a separate PR.

File: Snippets/Snippets_6/Snippets_6.csproj [hunk 1]
```diff
@@ -352,6 +352,23 @@
     <Compile Include="Transports\QueueCreation\QueueBindings.cs" />
     <Compile Include="HandlerOrdering.cs" />
     <Compile Include="Handlers\GenericAsyncHandler.cs" />
+    <Compile Include="UnitTesting\AdditionalServices\MyMessage.cs" />
+    <Compile Include="UnitTesting\AdditionalServices\MyService.cs" />
```
Comment (SimonCropp) [inline Snippets/Snippets_6/Snippets_6.csproj]: @petersgiles why is this here?

File: samples/sql-bridge/Version_6/MsmqPublisher/MsmqPublisher.csproj [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <UseVSHostingProcess>false</UseVSHostingProcess>
+  </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject />
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqPublisher/MsmqPublisher.csproj]: remove this

File: samples/sql-bridge/Version_6/MsmqPublisher/Program.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        while (true)
+        {
+            ConsoleKeyInfo key = Console.ReadKey();
+            Console.WriteLine();
+            Console.WriteLine("Press Enter to publish the SomethingHappened Event");
+
+            Guid eventId = Guid.NewGuid();
+            switch (key.Key)
+            {
+                case ConsoleKey.Enter:
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqPublisher/Program.cs]: for a single case lets make it an if

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/GuidBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+namespace MsmqToSqlRelay
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/GuidBuilder.cs]: think the namespace is redundant

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/ForwardMessagesToSqlBehavior.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
+﻿namespace MsmqToSqlRelay
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/ForwardMessagesToSqlBehavior.cs]: think the namespace is redundant

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/MsmqToSqlRelay.csproj [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <UseVSHostingProcess>false</UseVSHostingProcess>
+  </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject />
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/MsmqToSqlRelay.csproj]: think this is redundant

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/Program.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            Start(endpoint);
+        }
+        finally
+        {
+            await endpoint.Stop();
+        }
+    }
+
+
+    static void Start(IBusSession busSession)
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/Program.cs]: think we can inline this

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Iesi.Collections" version="4.0.1.4000" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="8.0.2" targetFramework="net452" />
+  <package id="NHibernate" version="4.0.4.4000" targetFramework="net45" />
+  <package id="NServiceBus" version="6.0.0-unstable1461" targetFramework="net452" />
+  <package id="NServiceBus.NHibernate" version="6.3.0-pullrequest0153" targetFramework="net452" />
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config]: we cant merge while it is still referencing a PR

File: samples/sql-bridge/Version_6/SqlRelay/Program.cs [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            Start(endpoint);
+        }
+        finally
+        {
+            await endpoint.Stop();
+        }
+    }
+
+    static void Start(IBusSession busSession)
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/SqlRelay/Program.cs]: inline this

File: samples/sql-bridge/Version_6/SqlSubscriber/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Reflection;
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/SqlSubscriber/Properties/AssemblyInfo.cs]: delete this

File: samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\Shared\Shared.csproj">
+      <Project>{77670cc2-dc46-4891-bc1f-a80f9f54bc48}</Project>
+      <Name>Shared</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj]: delete this

File: Snippets/Snippets_6/Snippets_6.csproj (hunk 1 above)
Comment (petersgiles) [inline Snippets/Snippets_6/Snippets_6.csproj]: checked in to this branch by mistake

File: samples/sql-bridge/Version_6/MsmqPublisher/MsmqPublisher.csproj (hunk 2 above)
Comment (petersgiles) [inline samples/sql-bridge/Version_6/MsmqPublisher/MsmqPublisher.csproj]: what am i removing? the startup object or the property group?

Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqPublisher/MsmqPublisher.csproj]: both

File: samples/sql-bridge/Version_6/SqlSubscriber/Properties/AssemblyInfo.cs (hunk 10 above)
Comment (petersgiles) [inline samples/sql-bridge/Version_6/SqlSubscriber/Properties/AssemblyInfo.cs]: do you mean the whole assemblyinfo file?

Comment (SimonCropp) [inline samples/sql-bridge/Version_6/SqlSubscriber/Properties/AssemblyInfo.cs]: yup

File: samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj (hunk 11 above)
Comment (petersgiles) [inline samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj]: are you talking about all the comment stuff

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config (hunk 8 above)
Comment (petersgiles) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config]: understood...merge blocked until then

File: samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj (hunk 11 above)
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/SqlSubscriber/SqlSubscriber.csproj]: yep

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config (hunk 8 above)
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config]: @petersgiles this looks like it is done https://github.com/Particular/NServiceBus.NHibernate/pull/153  do u want to merge and deploy it?

File: samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+  <package id="NServiceBus.SqlServer" version="3.0.0-pullrequest0151" targetFramework="net452" />
```
Comment (SimonCropp) [inline samples/sql-bridge/Version_6/MsmqToSqlRelay/packages.config]: can u try with NServiceBus.SqlServer 3.0.0-unstable0093 which is on myget

**Comment by indualagarsamy** [comment]
@SimonCropp - there was also a PR pending on the SqlTransport that uses the latest core. This sample won't work without those changes as well. https://github.com/Particular/NServiceBus.SqlServer/pull/150

**Comment by indualagarsamy** [comment]
@petersgiles - I am closing this PR. Since we decided to go with a more native approach. I've pushed the native sample in the msmqToSqlRelay branch. It works for v5. The v6 needs a more latest NSB.NH package for it to work. 

https://github.com/Particular/docs.particular.net/tree/msmqToSqlRelay/samples/msmqtosqlrelay

I'll open a PR as soon as the NServiceBus.NHibernate is on MyGet (the version which is compiled against the latest core). 

cc / @SimonCropp @HEskandari

**Comment by SimonCropp** [comment]
@indualagarsamy delete branch?

**Comment by indualagarsamy** [comment]
@SimonCropp - I was thinking of doing it after I get the other PR going :)
